offline/k8ssandra_setup.md

```diff
@@ -0,0 +1,201 @@
+# Setting up K8ssandra
+Reference - https://docs.k8ssandra.io/install/local/single-cluster-helm/
+
+K8ssandra will need following components to be installed in the cluster - 
+- dynamic persistent volume provisioning(for e.g with Openebs)
+- cert-manager
+- minio(for backup and restore)
+- K8ssandra-operator
+- Configure minio bucket for backups
+
+## [1] Dynamic Persistent Volume Provisioning
+If you already have a dynamic persistent volume provisioning setup, you can skip this step. Else, we will be using Openebs for dynamic persistent volume provisioning.
+
+Reference docs - https://openebs.io/docs/user-guides/local-storage-user-guide/local-pv-hostpath/hostpath-installation
+
+Deploy Openebs -
+
+```
+d helm install openebs charts/openebs --namespace openebs --create-namespace
+```
+The above helm chart will be readily available in the offline artifact.
+
+After successful deployment of openebs, you will see these storageclasses
+```
+d kubectl get sc
+NAME               PROVISIONER                    RECLAIMPOLICY   VOLUMEBINDINGMODE      ALLOWVOLUMEEXPANSION   AGE
+openebs-device     openebs.io/local               Delete          WaitForFirstConsumer   false                  5d20h
+openebs-hostpath   openebs.io/local               Delete          WaitForFirstConsumer   false                  5d20h
+```
+
+## [2] Install cert-manager
+Note: For wire-in-a-box setup, cert-manager pods should not be running on same node as ingress-nginx-controller. Hence, we should first install the ingress-nginx-controller and than install cert-manager.
+
+```
+cp ./values/ingress-nginx-controller/prod-values.example.yaml ./values/ingress-nginx-controller/values.yaml
+d helm install ingress-nginx-controller ./charts/ingress-nginx-controller --values ./values/ingress-nginx-controller/values.yaml
+```
+Now check where ingress-nginx-controller pod is running and install cert-manager on a different node.
+
+```
+d kubectl get pods -l app.kubernetes.io/name=ingress-nginx -o=custom-columns=NAME:.metadata.name,NODE:.spec.nodeName
+```
+For e.g. .. if it is `kubenode1`, taint the node
+```
+d kubectl cordon kubenode1
+```
+Now, download cert manager, and place it in the appropriate location:
+```
+wget https://charts.jetstack.io/charts/cert-manager-v1.13.2.tgz
+tar -C ./charts -xvzf cert-manager-v1.13.2.tgz
+```
+
+Install `cert-manager` into a new namespace `cert-manager-ns`.
+```
+d kubectl create namespace cert-manager-ns
+d helm upgrade --install -n cert-manager-ns --set 'installCRDs=true' cert-manager charts/cert-manager
+```
+
+Uncordon the node you cordonned earlier:
+```
+d kubectl uncordon kubenode1
+```
```

Comment:
Cert-manager is a must requirement for k8ssandra here - https://docs.k8ssandra.io/install/local/single-cluster-helm/#deploy-cert-manager

Currently, it is an optional step to use k8ssandra, that's why I put all the step at one place to avoid any confusions.
Once, this is default, plan is to install cert-manager and nginx-ingress-controller at the beginning and than install k8ssandra, to avoid duplication.

Reply:
I see. I think we should then reference the main `docs_ubuntu_22.04.md` part here rather than adding having this part twice in our docs.
If I follow the WIAB install, and want to deploy K8ssandra as well, I'd jump in here following the link right after installing the `wire-server` chart. This happens before the "Directing Traffic to Wire" section from the main readme, which tells us how to forward traffic to the ingress - even if it's not needed here, I'd argue we should keep the order as it is.
I would also jump back into the main readme after finishing the K8ssandra deployment and encounter errors by following the same ingress instructions again:
```
demo@Ubuntu-2204-jammy-amd64-base:~/wire-server-deploy$ d kubectl create namespace cert-manager-ns
Error from server (AlreadyExists): namespaces "cert-manager-ns" already exists
```

I guess everyone who made it this far is able to figure out that these steps are now redundant, but for the sake of consistency, I'd really rather see a reference to the relevant part in `docs_ubuntu_22.04.md` here, even if it's just a "temporary" change until we integrate K8ssandra into our default deployment.
Also, we don't have to keep track of changes in two places for the `cert-manager` part, if they come up.

Other opinions? @julialongtin perhaps?

Reply:
Okay this makes sense :) 
I have updated the docs to reference to existing docs for installing cert-manager.